Contributing.md

```diff
@@ -53,6 +53,68 @@ Contribution checklist
   instead of matching ``@first@`` and ``@second@``. You can use ``@.*?@`` in
   this case to stop early. The ``?`` tries to match _as few times_ as possible.
 
+* Beware of so-called "catastrophic backtracking".  Here is a badly written
+  regular expression to match strings:
+
+  ```python
+  r'"(\\?.)*?"'
```

Comment:
This is a more non-obvious example, which is good, but you could additionally give the "easy one" with repeated repetition, like `(a*)+`.